### DISCUSSION TITLE
Pulse Height Correction for low sample rates, is it possible?

### DISCUSSION THREAD

**Discussion by ssesselmann** [ssesselmann/impulse#14] [Ideas]
On the attached images generated using an electronic pulser at 40 kHz we can see the effect of shape tolerance on resolution. 

Essentially we observe that the pulse height is always **under sampled** never **over sampled**, and we can clearly see that there is a correlation between distortion and the amount of under sampling. 

It should therefore be possible to apply a small correction factor to the pulse height based on the amount of distortion measured.

In these examples we can see that a distortion of 4000 or more can mean up to 5% pulse height loss

![1000](https://github.com/ssesselmann/impulse/assets/75355962/6ac62e20-a214-4988-8ea2-d54db418175b)
![2000](https://github.com/ssesselmann/impulse/assets/75355962/293f3666-7b04-43d2-b8f2-77a9ee84c11e)
![3000](https://github.com/ssesselmann/impulse/assets/75355962/28a2dab1-7fdc-4468-a443-3a20773f2d9a)
![4000](https://github.com/ssesselmann/impulse/assets/75355962/b0087ee6-c15a-430b-b87e-5a0d99289f04)

**Comment by BOPOHOP** [comment]
Interesting... What does the collected pulse shape look like?

**Reply by ssesselmann** [reply]
It looks like a regular 48 kHz pulse, the only difference is I am using a synthetic pulse electronically generated, so the resolution is around 0.5%. So as you know if the true peak falls between two samples the full pulse height will not be realised. What we see here is a correlation between distortion and under sampling.
